source/docs/casper/resources/tutorials/advanced/cross-contract.md

```diff
@@ -0,0 +1,538 @@
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Cross Contract Communication
+
+Level of the tutorial: `Advanced`
+
+Time to complete the tutorial: `45 - 60 Minutes`
+
+Important topics to read before going further:
+
+- [getting started](../../../developers/writing-onchain-code/getting-started.md)
+- [writing on-chain code](../../../developers/writing-onchain-code/simple-contract.md)
+- [calling contracts](../../../developers/writing-onchain-code/calling-contracts.md)
+
+This will assure that you have needed tools and base knowledge to complete this tutorial. If not everything was clear while reading these sections, don't worry as we will be doing a full walkthrough step by step.
+
+## Outline of the tutorial {#outline}
+
+This tutorial covers all possible variations of cross contract communication. While it not necessary when writing a simple contract which doesn't have to interact with other deployed contracts it is nontheless important to understand the ways to achieve this feat.
```

Comment:
The second sentence here doesn't make sense. of course learning how to communicate between contracts is not necessary when writing contracts that don't communicate That is by definition. The clue is in the name. But then you say that it is important to know how to do it without saying WHY it is important. The whole paragraph is unneccesary and doesn't tell me anything.

I think what you were trying to say was something like "Of course, it is possible to write a contract that doesn't use cross-contract communication, but most contracts do, so you really should learn this". Am I reading this right?

Reply:
Rewritten this section to be more concise